Back-To-School/Assets/BackToSchool/Scripts/Enemies/EnemyWarrior.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections;
+using Assets.BackToSchool.Scripts.Interfaces;
+using UnityEngine;
+
+
+namespace Assets.BackToSchool.Scripts.Enemies
+{
+    public class EnemyWarrior : BaseEnemy, IMeleeAttackable
+    {
+        [SerializeField] private float _delayBeforeDamage = 0.5f;
+        public void DoDamage(float damage) => _target.GetComponent<IDamageable>().TakeDamage(damage);
```

Comment:
Hmm. Is that ok that player would receive damage in any case even he `dash` (fastly run or sth like) away from hit?